SpatialGDK/Source/SpatialGDK/Private/Interop/ActorSystem.cpp

```diff
@@ -1539,17 +1571,10 @@ void ActorSystem::RemoveActor(const Worker_EntityId EntityId)
 	// Cleanup pending add components if any exist.
 	if (USpatialActorChannel* ActorChannel = NetDriver->GetActorChannelByEntityId(EntityId))
 	{
-		// If we have any pending subobjects on the channel
+		// If we have any pending subobjects on the channel, remove them
 		if (ActorChannel->PendingDynamicSubobjects.Num() > 0)
 		{
-			// Then iterate through all pending subobjects and remove entries relating to this entity.
-			for (const auto& Pair : PendingDynamicSubobjectComponents)
-			{
-				if (Pair.Key == EntityId)
-				{
-					PendingDynamicSubobjectComponents.Remove(Pair);
-				}
-			}
+			PendingDynamicSubobjectComponents.Remove(EntityId);
```

Comment:
Is the order important, or could we use RemoveSwap?

Reply:
This is a TMap

Reply:
And again!